### DISCUSSION TITLE
How to change variable implemented in SU2

### DISCUSSION THREAD

**Discussion by jjywow** [su2code/SU2#1924] [Q&A] [answered]
Hello, 

I'm changing the airfoil shape with SU2_DEF using the Hicks-Henne function.

It works well but I like to change the variable value defined in SU2. 

For Hicks-Henne function is as below, and I know 't' value in SU2 is defined as 3.0. 
 z= z+ ( sin(..~..) ) ^t

I read some papers that change the value t in SU2. 

Where can I change the value?

I tried to change the t value by changing 'const su2double t2 = 3.0' to 'const su2double t2 = 10.0'  in the file '\SU2\Common\src\grid_movement\CSurfaceMovement.cpp', but it didn't work. Maybe I guess it should be a different file to modify.

Please give me some advice on this. Even if you are using a different function to change, could you share where to modify the predefined variable values?

Thanks

**Comment by bigfooted** [comment, marked as answer]
You can check in vscode if you access this routine, and also what the value of fk will be after the change.
So I would run it in debug mode first to make sure that this is the t2 you're looking for.

**Reply by jjywow** [reply]
Thanks for your help!
So, do you think  '\SU2\Common\src\grid_movement\CSurfaceMovement.cpp' is the right file for changing variable t for the Hicks-Henne bump function?
vscode means visual studio, right? debug mode in visual studio or SU2? sorry I'm not good at these stuff could you elaborate on how to debug?

I was kind of sure about the t2, though. Because as you can see, t2 appears only four times in the file.

  const su2double t2 = 3.0;

  if (config->GetParamDV(iDV, 0) == NO) { upper = false; }
  if (config->GetParamDV(iDV, 0) == YES) { upper = true; }

  for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {

    for (iVertex = 0; iVertex < boundary->nVertex[iMarker]; iVertex++) {
      VarCoord[0] = 0.0; VarCoord[1] = 0.0; VarCoord[2] = 0.0;

      if (config->GetMarker_All_DV(iMarker) == YES) {

        Coord_ = boundary->vertex[iMarker][iVertex]->GetCoord();
        Normal_ = boundary->vertex[iMarker][iVertex]->GetNormal();

        /*--- The Hicks Henne bump functions should be applied to a basic airfoil without AoA,
         and unitary chord, a tranformation is required ---*/

        ValCos = cos(AoA*PI_NUMBER/180.0);
        ValSin = sin(AoA*PI_NUMBER/180.0);

        Coord[0] = Coord_[0]*ValCos - Coord_[1]*ValSin;
        Coord[0] = max(0.0, Coord[0]); // Coord x should be always positive
        Coord[1] = Coord_[1]*ValCos + Coord_[0]*ValSin;

        Normal[0] = Normal_[0]*ValCos - Normal_[1]*ValSin;
        Normal[1] = Normal_[1]*ValCos + Normal_[0]*ValSin;

        /*--- Bump computation ---*/

        ek = log10(0.5)/log10(xk);
        if (Coord[0] > 10*EPS) fk = pow( sin( PI_NUMBER * pow(Coord[0], ek) ), t2 );

**Reply by jjywow** [reply]
After modifying the CSurfaceMovement.cpp, should I re-compile the SU2??

**Reply by bigfooted** [reply]
Yes, you should compile using` ./meson.py build -Ddebug=false -Doptimization=0 -Dwith-mpi=disabled ...`
and then in Visual Studio Code in the Debug session you can add a configuration, choose c/c++ (gdb) Launch and fill in the details of
```

"program":"/path/to/SU2_CFD" 
"args":"config.cfg" 
"cwd": "/path/to/config"

```

Then see if you can run it in debug mode from within VSCode. If it works, you can place a breakpoint in the function to make it stop where you want (just click next to the line number and a red circle will appear to indicate a breakpoint). You can then inspect the values of the variables.

**Reply by jjywow** [reply]
Thank you very much! It worked properly. Problem resolved. Have a good day
